rlclientlib/vw_model/safe_vw.cc

```diff
@@ -350,9 +371,22 @@ safe_vw_factory::safe_vw_factory(const model_management::model_data&& master_dat
   : _master_data(master_data)
   {}
 
-  safe_vw* safe_vw_factory::operator()()
-  {
-    if (_master_data.data())
+safe_vw_factory::safe_vw_factory(const model_management::model_data& master_data, const std::string& command_line)
+  : _master_data(master_data), _command_line(command_line)
+  {}
+
+safe_vw_factory::safe_vw_factory(const model_management::model_data&& master_data, const std::string& command_line)
```

Comment:
When this constructor is called I can see we do `safe_vw_factory(std::move(data))` in vw_model.cc. To take advantage of actually moving and not copying the data the constructor should remove `const` (i.e. become `safe_vw_factory(model_management::model_data&& master_data, ...)` and also `master_data` it should be moved into `_master_data` (i.e. line 379 `_master_data(std::move(master_data))` )

Reply:
Thanks Olga. That is great feedback. Here, I have copied the pattern from other constructor. I do not have much experience with c++ to be able to confidently make that change though.

Reply:
Ah we do the same, probably by mistake. No worries I can patch it right after and tag you so that you see the change?